src/example.c

```diff
@@ -79,12 +79,23 @@ int main(void)
     //                               "application.log");
 
     sentry_init(options);
-    send_event();
-
-    sentry_shutdown();
 
-    // normally happens before sentry_shutdown
-    startup();
+    // `make run message`
+    if( strncmp(argv[1], "--crash", 4) ) {
```

Comment:
There are three small issues in this line:
 - `strncmp` only compares `4` characters in this case, so you could also pass `--cringe` and it would compare. Instead, just use `strcmp`.
 - `strcmp` returns `0` when the strings match, which effectively evaluates to false. But that means if I pass *anything* that doesn't match `--crash`, it would execute this branch, such as `--unknown`. This is rather confusing control flow. Instead, you could compare the return value to `0` and  then execute the crashing instruction.
- This line will crash (unintentionally :)), or execute random behavior, if no arguments are passed to the program. Instead, you can check `argc` before.

Together, this would become:

```suggestion
    if (argc != 2) {
        printf("Usage: %s [--crash|--message]\n", argv[0]);
    } else if (strcmp(argv[1], "--crash", 7) == 0) {
        startup();
        // ...
```

Reply:
I follow. The first time example of strcmp I tried involved passing some 'buffer[80]' as an argument so it didn't work out. This one is better.